Twig/DoctrineExtension.php

```diff
@@ -305,15 +305,35 @@ function ($matches) use ($parameters, &$i) {
             $query
         );
 
-        if ($highlight) {
-            $result = \SqlFormatter::highlight($result);
-            $result = str_replace(array('<pre ', '</pre>'), array('<span ', '</span>'), $result);
-        }
-
         return $result;
     }
 
     /**
+     * Highlights the syntax of the given SQL statement.
+     *
+     * @param  string $sql
+     * @return string
+     */
+    public function highlightSqlSyntax($sql)
+    {
+        \SqlFormatter::$pre_attributes = 'class="highlight highlight-sql"';
+        \SqlFormatter::$quote_attributes = 'class="string"';
+        \SqlFormatter::$backtick_quote_attributes = 'class="string"';
+        \SqlFormatter::$reserved_attributes = 'class="keyword"';
+        \SqlFormatter::$boundary_attributes = 'class="symbol"';
+        \SqlFormatter::$number_attributes = 'class="number"';
+        \SqlFormatter::$word_attributes = 'class="word"';
+        \SqlFormatter::$error_attributes = 'class="error"';
+        \SqlFormatter::$comment_attributes = 'class="comment"';
+        \SqlFormatter::$variable_attributes = 'class="variable"';
+
+        $html = \SqlFormatter::highlight($sql);
+        $html = preg_replace('/<pre class="(.*)">(.*)<\/pre>/Us', '<div class="\1"><pre>\2</pre></div>', $html);
```

Comment:
``(.*)`` should be ``([^"]*+)`` to catching more than necessary here (in case there is nested tags inside the pre with their own attributes). The ``+`` in my proposal is optimizing the regex by making the quantifier possessive, which is possible with my proposal (as the quantifier cannot match the ``"`` following it)